src/components/utils/include/utils/timer.h

```diff
@@ -84,47 +90,127 @@ class Timer {
    */
   bool IsRunning() const;
   /**
-   * @brief GetTimeout
-   * @return returns timeout
+   * @brief Suspends timer execution after next loop.
    */
-  Milliseconds GetTimeout() const;
-
- private:
-  const std::string name_;
-  const TimerTask* task_;
-  bool repeatable_;
-  uint32_t timeout_ms_;
-  bool is_running_;
-  timer_t timer_;
-  mutable sync_primitives::Lock lock_;
-  sync_primitives::Lock task_lock_;
+  virtual void Suspend();
 
   /**
-   * @brief method called from friend handler_wrapper and call run() from task.
+   * @brief Stop timer update timeout and start timer again
+   * Note that it cancel thread of timer, If you use it from callback,
+   * it probably will stop execution of callback function
+   * @param timeout_milliseconds new timeout value
+   *
    */
-  void OnTimeout();
+  virtual void UpdateTimeOut(const uint32_t timeout_milliseconds);
```

Comment:
@anosach-luxoft , `Milliseconds ` ok, `virtual `stay from old logic, if you want I'll delete it.